qiskit_experiments/library/calibration/fine_drag.py

```diff
@@ -0,0 +1,274 @@
+        Inverting the relation above yields the error in :math:`\beta` that produced the rotation
+        error :math:`{\rm d}\theta`.
+
+        .. math::
+
+            {\rm d}\beta=\frac{\sqrt{\pi}\,{\rm d}\theta\sigma}{ \theta_\text{target}^2}
+
+    # section: see_also
+        qiskit_experiments.library.calibration.drag.DragCal
+
+    # section: reference
+        .. ref_arxiv:: 1 1612.00858
+        .. ref_arxiv:: 2 1011.1949
+    """
+
+    __analysis_class__ = FineDragAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine amplitude experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that Rp - Rm gate sequence
+                is repeated.
+            schedule (ScheduleBlock): The schedule for the plus rotation.
+            normalization (bool): If set to True the DataProcessor will normalized the
+                measured signal to the interval [0, 1]. Defaults to True.
+            add_sx (bool): If True an sx gate is pre-pended to the circuit.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(20))
+        options.schedule = None
+        options.add_sx = None
+
+        return options
+
+    @classmethod
+    def _default_analysis_options(cls) -> Options:
+        """Default analysis options."""
+        options = super()._default_analysis_options()
+        options.normalization = True
+        options.angle_per_gate = 0.0
+        options.phase_offset = -np.pi / 2
+        options.amp = 1.0
+
+        return options
+
+    def __init__(self, qubit: int):
+        """Setup a fine amplitude experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine amplitude calibration experiment.
+        """
+        super().__init__([qubit])
+
+    @staticmethod
+    def _pre_circuit() -> QuantumCircuit:
+        """Return the quantum circuit to apply before repeating the Rp - Rz - Rp - Rz gates."""
+        return QuantumCircuit(1)
+
+    @staticmethod
+    def _post_circuit() -> QuantumCircuit:
+        """Return the quantum circuit to apply after repeating the Rp - Rz - Rp - Rz gates."""
+        return QuantumCircuit(1)
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the fine DRAG calibration experiment.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            A list of circuits with a variable number of gates. Each gate has the same
+            pulse schedule.
+        """
+        schedule, circuits = self.experiment_options.schedule, []
+
+        drag_gate = Gate(name=schedule.name, num_qubits=1, params=[])
+
+        for repetition in self.experiment_options.repetitions:
+            circuit = self._pre_circuit()
+
+            if self.experiment_options.add_sx:
```

Comment:
I think this can go to pre circuit method, and we can remove user-unconformable option (`add_sx`, i.e. this is tied to experiment type) by hard-cording this sx gate in the pre circuit method of a specific subclass. 

If user change this, I guess analysis will be broken. Such thing should not go to options.

Reply:
Fair enough. See here: 6a83ea7